### PR TITLE
chore(tools): use ESLint class

### LINKED ISSUE #1215: [tools] use ESLint class instead of CLIEngine
Since version 7.0.0, ESLint supports `ESLint` class instead of `CLIEngine` class.
https://github.com/eslint/eslint/pull/12939
https://github.com/eslint/rfcs/pull/40

So, if it's possible, let's use `ESLint` class!

### PR DESCRIPTION
**Pull request by tyankatsu0105**
fix #1215

### CODE REVIEW

**Comment by tyankatsu0105** [comment]
All tests failed... why??🤔

**Comment by ota-meshi** [comment]
Probably using ESLint v7.3 seems to fail some test cases. I think this is because there are some test cases with empty errors in the indented test cases.

I will fix this.

**Comment by ota-meshi** [comment]
@tyankatsu0105 I fixed test cases. Could you rebase this PR?

File: tools/update-lib-configs.js [hunk 1]
```diff
@@ -88,6 +88,10 @@ categories.forEach((category) => {
 })
 
 // Format files.
-const linter = new eslint.CLIEngine({ fix: true })
-const report = linter.executeOnFiles([ROOT])
-eslint.CLIEngine.outputFixes(report)
+async function main() {
```
Comment (ota-meshi) [inline tools/update-lib-configs.js:91]: This process is not the "main" of this script, so can you rename it to "format" etc.?

Comment (tyankatsu0105) [reply tools/update-lib-configs.js]: done

**Pull request merged**